frontend/src/metabase/visualizations/echarts/cartesian/scatter/model/index.ts

```diff
@@ -0,0 +1,158 @@
+      seriesModel.bubbleSizeDataKey != null
+        ? seriesModel.bubbleSizeDataKey
+        : null,
+    )
+    .filter(isNotNull);
+
+  if (bubbleSizeDataKeys.length === 0) {
+    return null;
+  }
+
+  const bubbleSizeMaxValues = getObjectValues(
+    getDatasetExtents(bubbleSizeDataKeys, dataset),
+  ).map(extent => extent[1]);
+  const bubbleSizeDomainMax = Math.max(...bubbleSizeMaxValues);
+
+  return [0, bubbleSizeDomainMax];
+};
+
+export function getScatterPlotModel(
+  rawSeries: RawSeries,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+  showWarning?: ShowWarning,
+): ScatterPlotModel {
+  // rawSeries has more than one element when two or more cards are combined on a dashboard
+  const hasMultipleCards = rawSeries.length > 1;
+  const cardsColumns = getCardsColumns(rawSeries, settings);
+  const columnByDataKey = getCardsColumnByDataKeyMap(rawSeries, cardsColumns);
+  const dimensionModel = getDimensionModel(rawSeries, cardsColumns);
+  const unsortedSeriesModels = getCardsSeriesModels(
+    rawSeries,
+    cardsColumns,
+    settings,
+    renderingContext,
+  );
+
+  // We currently ignore sorting and visibility settings on combined cards
+  const seriesModels = hasMultipleCards
+    ? unsortedSeriesModels
+    : getSortedSeriesModels(unsortedSeriesModels, settings);
+
+  const unsortedDataset = getScatterPlotDataset(rawSeries, cardsColumns);
+  const dataset = sortDataset(
+    unsortedDataset,
+    settings["graph.x_axis.scale"],
+    showWarning,
+  );
+
+  const xAxisModel = getXAxisModel(
+    dimensionModel,
+    rawSeries,
+    dataset,
+    settings,
+    renderingContext,
+    showWarning,
+  );
+  const yAxisScaleTransforms = getAxisTransforms(
+    settings["graph.y_axis.scale"],
+  );
+
+  const transformedDataset = applyVisualizationSettingsDataTransformations(
+    dataset,
+    [],
+    xAxisModel,
+    seriesModels,
+    yAxisScaleTransforms,
+    settings,
+    showWarning,
+  );
+
+  const { formatters: seriesLabelsFormatters, compactSeriesDataKeys } =
+    getSeriesLabelsFormatters(
+      seriesModels,
+      [],
+      transformedDataset,
+      settings,
+      renderingContext,
+    );
+
+  const { leftAxisModel, rightAxisModel } = getYAxesModels(
+    seriesModels,
+    transformedDataset,
+    settings,
+    columnByDataKey,
+    false,
+    [],
+    compactSeriesDataKeys,
+    renderingContext,
+  );
+
+  const trendLinesModel = getTrendLines(
+    rawSeries,
+    [leftAxisModel, rightAxisModel],
+    yAxisScaleTransforms,
+    seriesModels,
+    transformedDataset,
+    settings,
+    [],
+    renderingContext,
+  );
+
+  return {
+    stackModels: [],
+    dataset,
+    transformedDataset,
+    seriesModels,
+    yAxisScaleTransforms,
+    columnByDataKey,
+    dimensionModel,
+    xAxisModel,
+    leftAxisModel,
+    rightAxisModel,
+    trendLinesModel,
+    bubbleSizeDomain: getBubbleSizeDomain(seriesModels, transformedDataset), // TODO move function
```

Comment:
@alxnddr Can you remove the TODO comment before merging this? By mistake I left this in my original PR. I have a PR open to remove this from master https://github.com/metabase/metabase/pull/43423